utils/global-state-update-gen/README.md

```diff
@@ -0,0 +1,106 @@
+- with `only_listed_validators` set to false, we need only a single `[[accounts]]` entry for the validator we want to change,
+- with `only_listed_validators` set to true, we need 100 `[[accounts]]` entries, one per each account that is supposed to be a validator after the upgrade.
+
+On the other hand, replacing 100 validators with 5 different ones (a use case in testing setups) would require:
+- just 5 entries for the new validators if `only_listed_validators` is true,
+- 105 entries - 100 to remove the old validators, and 5 to add the new ones - if `only_listed_validators` is false.
+
+The `[[transfers]]` definitions simply transfer funds from one account to another. Every definition requires a source account, a target account and an amount to be defined. If the source account doesn't contain enough funds for the transfer, it won't be executed. If the target account doesn't exist, it will be created.
+
+**Note:** transfers are executed before the `[[accounts]]` definitions. This means that it is possible to overwrite the effects of a transfer if the source or target account is also specified among `[[accounts]]`.
+
+After the transfers have been executed and account balances and stakes have been updated, the tool also updates the auction contract state in the form of the snapshot of validators sets for the next few eras, as well as bids and withdraws:
+- bids of accounts with zero stake are set to empty, and bids of accounts with nonzero stake are created or updated to reflect the configured amounts,
+- if `only_listed_validators` is true, any bid larger than the smallest stake among the new set of validators is reset to zero,
+- the withdraws of validators that are being removed from the set are cancelled.
+
+The tool also takes care to update the total supply in the network to reflect the changes in balances resulting from the configured modifications to the state.
+
+### Legacy commands
+
+#### `validators`
+
+Usage: `global-state-update-gen validators -d DATA-DIRECTORY -s STATE-ROOT-HASH -v VALIDATOR-KEY,STAKE,BALANCE -v VALIDATOR-KEY,STAKE,BALANCE ...`
+
+Apart from the common `-d` and `-s` parameters, the subcommand has one additional parameter, `-v` or `--validator`. Multiple such parameters can be supplied. Also note that the third field, `BALANCE`, is optional (ie., the definition can be just `-v VALIDATOR-KEY,STAKE`).
+
+Every `-v` instance configures a single validator to be included in the set after the upgrade. A `-v KEY,STAKE,BALANCE` corresponds to an `[[accounts]]` entry in the config file:
+
+```toml
+[[accounts]]
+public_key = "KEY"
+stake = "STAKE"
+balance = "BALANCE"
+```
+
+The command as a whole works just like a config file with only `[[accounts]]` entries and `only_listed_validators` set to `true`.
+
+#### `balances`
+
+Usage: `global-state-update-gen balances -d DATA_DIRECTORY -s STATE_ROOT_HASH -f FROM-ACCOUNT -t TO-ACCOUNT -a AMOUNT`
```

Comment:
`transfer` might make more sense for this command.

Reply:
True - but it has been this way for some time now, and even though I don't think anyone besides us has been using it, I'm not sure if we should be changing the user interface this much.